module/model-elasticsearch/src/util.ts

```diff
@@ -201,7 +201,7 @@ export class ElasticsearchUtil {
       if (o[x] === undefined || o[x] === null) {
         ops.push(`ctx._source.${path}${path ? '.' : ''}remove("${x}")`);
       } else if (Util.isPrimitive(o[x]) || Array.isArray(o[x])) {
-        const param = prop.replace(/[^A-Za-z_$]/g, '_');
+        const param = prop.replace(/[^a-z0-9_$]/g, '_');
```

Comment:
If you are dropping uppercase letters in the check, you will need to call `.toLowerCase` on the prop before running replace.  Otherwise capital letters will be dropped and could lead to problems.

Reply:
Agreed, this is a better way to handle upper case than replacing with underscore. PR updated